wrangler-core/src/main/java/io/cdap/directives/column/ExplodeRecord.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.directives.column;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.wrangler.api.Arguments;
+import io.cdap.wrangler.api.Directive;
+import io.cdap.wrangler.api.DirectiveExecutionException;
+import io.cdap.wrangler.api.DirectiveParseException;
+import io.cdap.wrangler.api.ExecutorContext;
+import io.cdap.wrangler.api.Pair;
+import io.cdap.wrangler.api.Row;
+import io.cdap.wrangler.api.annotations.Categories;
+import io.cdap.wrangler.api.lineage.Lineage;
+import io.cdap.wrangler.api.lineage.Many;
+import io.cdap.wrangler.api.lineage.Mutation;
+import io.cdap.wrangler.api.parser.Bool;
+import io.cdap.wrangler.api.parser.ColumnName;
+import io.cdap.wrangler.api.parser.ColumnNameList;
+import io.cdap.wrangler.api.parser.Text;
+import io.cdap.wrangler.api.parser.TokenType;
+import io.cdap.wrangler.api.parser.UsageDefinition;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A directive for splitting a RowType ("Record") Column into multiple Columns.
+ */
+@Plugin(type = Directive.TYPE)
+@Name(ExplodeRecord.NAME)
+@Categories(categories = { "column"})
+@Description("Flattens a record into individual columns.")
+public class ExplodeRecord implements Directive, Lineage {
+  public static final String NAME = "explode-record";
+  // Column on which to apply mask.
+  private String[] columns;
```

Comment:
There's already a directive called `Flatten` that works for arrays. That directive takes a list of columns. Making a different syntax for two similar directives might confuse the users even more.

Reply:
I still think this should just operate on a single column, it doesn't make sense for this to operate on multiple columns, otherwise all the directives related to single column should be extended to be multiple columns. We shouldn't use old wrong behavior as tradition. With new implementation, we should make sure things are correct

Reply:
I think we should actually support multiple columns. The key is that we don't support many transformations on nested data. You *have* flatten them. So providing an option to do it in a single directive does seem more user friendly.